MarkUsProject/markus-autotesting
# haskell tester: make stack resolver version a setting

Make Stack Resolver a setting. It is a setting in two places. One is in the environment setting, where when we first install the tester, we will install this version in the setting. Second, is on the test settings where the instructor decides on the version. Currently there is only 1 resolver version to choose from. In the future, we will add to the list.

Note: 
in the file "server/autotest_server/__init__.py" we added the following lines:

                requirements_path = os.path.join(os.path.dirname(__file__), "../requirements.txt")
                subprocess.run([f"{default_env}/bin/pip", "install", "-r", requirements_path], check=True)

The reason for this is because when we import the config for the Haskell tester, we require the jsonschema module. This was not already present in the defaultvenv. 

2 Testing procedure:
1) Environment Setting: 
   - requires rebuilding docker environment for new env var STACK_RESOLVER
   - save Automtated Testing settings to see if that is successful
   - run a test for a submission to see if that is successful
 2) Test setting:
   - Rebuild schema: we have a new field "Resolver version" 
   - Save the Haskell Automated Test settings to include the Resolver version
   - Run a submission for a Haskell assignment to see if it is successful
